### ISSUE TITLE
Should we keep From.useForm() returning the same value? 是否需要保持 useForm 创建的对象不变？

### ISSUE THREAD

**Issue by HerbLuo** [ant-design/ant-design#20550]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-7qxbr)

### Steps to reproduce
I want to set form value after a network request, and i use `useEffect` hooks api, and then pass an array as the second argument to avoid re-running.
And when i pass the `[]` as the second argument, react warns `React Hook React.useEffect has a missing dependency: 'form'`,
So i pass the `[form]` as the argument, but the effect run again may because the `form` is not a fixed value.

And when the `Foo` is re-running (change the text value will trigger it), the text value will reset by remote server.

### What is expected?
The effect will be skipped.

### What is actually happening?
The effect was called several times.

| Environment | Info |
|---|---|
| antd | 4.0.0-beta.1 |
| React | 16.12.0 |
| System | Manjaro Linux |
| Browser | Chrome 79 |

**Comment by cbingrui** [comment]
@HerbLuo hello, it this an issue? how can you make the Form.useForm be used in react hooks?

**Comment by yoyo837** [comment]
```js
import { Form} from 'antd';

export default () => {
  const [form] = Form.useForm();
};
```

**Comment by soulcm** [comment]
Why this issue had been closed?  the 4.0 version also has the same problem, Form.useForm() didn't return the same value
